src/auxinfo/participant.rs

```diff
@@ -331,10 +331,12 @@ impl AuxInfoParticipant {
     ) -> Result<ProcessOutcome<<Self as ProtocolParticipant>::Output>> {
         info!("Handling round one auxinfo message.");
 
-        let message = broadcast_message.extract_message(BroadcastTag::AuxinfoR1CommitHash)?;
+        let message = broadcast_message
+            .clone()
```

Comment:
Oh, hm. The goal of taking this by value is to avoid cloning or, more generally, copies of the type floating around longer than we need them. It's not security critical in this case but that is the design motivation. Here I guess we'd have to go up and see if we can pass the `BroadcastOutput` by value to the `handle_round_one_msg` method.

I think that will go bother the slightly complicated `convert` method as well, but I think it should compile. Happy to walk through the fancy types in that function next week, if you'd like.

Reply:
I see. Sure we can go through it together either today or tomorrow if you are free